docs/devel/networking.md

```diff
@@ -205,3 +205,136 @@ to the user.
+  networks:
+  - name: masqueradenet
+    pod:
+      vmNetworkCIDR: 10.11.12.0/24
+```
+
+As with bridge binding, the `discoverPodNetworkInterface` caches the MTU of
+the pod networking interface. It also assigns an IP address from the configured
+CIDR (or the default), and reserves an IP for the gateway - which will be the
+in-pod bridge. Re-using the example above, we would have `10.11.12.1` as
+gateway IP, and `10.11.12.2` as VM IP.
+
+NAT is configured in the masquerade `preparePodNetworkInterfaces` method. In
+order to prepare for that, a dummy interface is created, and is set as a slave
+of the in-pod bridge. The bridge is configured with the IP address which was
+previously reserved for the VM's gateway.
+
+Afterwards, the nftables / iptables rules are provisioned in the NAT table. It
+follows a standard one to one NAT implementation using netfilter, but let's
+look a bit further on the details.
+
+It first involves the `PREROUTING` chain, which is responsible for packets that
+have just arrived at the network interface. This rule simply filters all
+incoming traffic from the pod networking interface, making it go through the
+`KUBEVIRT_PREINBOUND` chain.
+
+```
+Chain PREROUTING (policy ACCEPT)
+target               prot opt source               destination
+KUBEVIRT_PREINBOUND  all  --  anywhere             anywhere
+```
+
+On the `KUBEVIRT_PREINBOUND` chain, packets will be DNAT'ed (have their
+destination address changed) to the IP address of the VM - e.g. `10.11.12.2`.
+This can be subject to a whitelist of ports (if one was provided by the user,
+in the masquerade interface specification) or simply have all ports accepted -
+when the port configuration is omitted.
+
+```
+Chain KUBEVIRT_PREINBOUND (1 references)
+target     prot opt source               destination
+DNAT       tcp  --  anywhere             anywhere             tcp dpt:http to:10.0.2.2
+```
+
+Before the packet leaves the interface, it will pass through the
+`POSTROUTING` chain, which in turn, makes the packet go through the
+`KUBEVIRT_POSTINBOUND` chain.
+
+```
+Chain POSTROUTING (policy ACCEPT)
+target     prot opt source               destination
+KUBEVIRT_POSTINBOUND  all  --  anywhere   anywhere
+```
+
+In the `KUBEVIRT_POSTINBOUND` chain, SNAT is performed: the source IP address
+of the outbound packet is modified to the IP address of the gateway -
+`10.11.12.1`.
+
+```
+Chain KUBEVIRT_POSTINBOUND (1 references)
+target     prot opt source               destination
+SNAT       tcp  --  anywhere             anywhere             tcp dpt:http to:10.0.2.1
+```
+
+Once the packet leaves the interface, it will be subject to the routing tables
+present on the virt-launcher pod. Since we've DNAT'ed the packet, it will be
+routed to the in-pod bridge, which will forward the traffic to the VM. The
+route responsible for routing the traffic from the pod interface to the in-pod
+bridge is:
+
+```
+10.0.2.0/24 dev k6t-eth0 proto kernel scope link src 10.0.2.1
```

Comment:
It is strange that you explicitly mention this directly connected route. As if you imply there is something special about it (like one needs to add it explicitly).

Reply:
Removed it; it'll make more sense for the ipv6 section.